### ISSUE TITLE
Keep stable branches based on CDash status

### ISSUE THREAD

**Issue by ibaned** [gahansen/Albany#218]
This issue is for us to start automatically monitoring the CDash status and using that to update stable branches/forks of Trilinos and Albany. This is an evolution of the "last known good versions" that @lxmota used to maintain.

I've come up with an idea, which is to use the HTML content from the CDash site directly to determine the status of things.

I've pushed the script here:

https://github.com/gahansen/Albany/blob/master/tools/cdash_all.py

Although it does require a build of Python 3 on a Sandia machine with the "Beautiful Soup" package installed.

When run with an appropriate Python 3 install it produces something like this:

```
13:30:07:cee-compute011:~/src/cdash_all$ ~/install/python3/bin/python3 cdash_all.py
Configure not reported for Trilinos_SEMS_cee-build012_serial-gcc-release
Build not reported for Trilinos_SEMS_cee-build012_serial-gcc-release
JengaAlbany had 2 test failures
Albany32Bit had 7 test failures
Albany64BitClang had 3 test failures
Albany32Bit had 4 test failures
AlbanyIntel had 10 test failures
Albany64BitClang had 5 test failures
albany_SEMS_ceerws1006_serial-intel-release had 1 test failures
albany_cluster-toss3_skybridge-login5_serial-intel-release had 1 test failures
```

We can build on this tool to use it as a green-light check in a higher-level script that updates stable branches

**Comment by bartgol** [comment]
So you would put this in a master script, and, if cdash_all returns ok, the master script proceeds to update trilinos? Sounds good.

**Comment by ibaned** [comment]
@bartgol yes, thats the plan. I think we should interpret all test failures as a problem, so now the new effort is to fix all the remaining failing tests, or remove them.
